drivers/gpio/gpio_sch.c

```diff
@@ -128,6 +131,12 @@ static int gpio_sch_config(struct device *dev,
 {
 	const struct gpio_sch_config *info = dev->config->config_info;
 
+	/* Do some sanity check first */
+	if (flags & (GPIO_INT | GPIO_INT_LEVEL)) {
```

Comment:
@dcpleung @galak @tbursztyka I'm reasonably sure this doesn't do what you think it does, because `GPIO_INT_LEVEL` is zero.  So I'm pretty sure you've made it so that you're exiting with `-EINVAL` if interrupts are enabled.

There were commits in #11880 which addressed several similar bugs (esp32 in particular is also broken).

Reply:
Thanks for letting me know. I have submitted #13143 to fix this.